File: mpi-intro/minmeanmax.f90
       program randomdata
       implicit none
       integer,parameter :: nx=1500
       real, allocatable :: dat(:)

       integer :: i,n
       real    :: datamin, datamax, datamean

!
! random data
!
       allocate(dat(nx))
       call random_seed(size=n)
       call random_seed(put=[(i,i=1,n)])
       call random_number(dat)
       dat = 2*dat - 1.

!
! find min/mean/max
! 
       datamin = minval(dat)
       datamax = maxval(dat)
       datamean= (1.*sum(dat))/nx

       deallocate(dat)

       print *,'min/mean/max = ', datamin, datamean, datamax
  
       return
       end


  

File: mpi-intro/minmeanmax.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    const int nx=1500;
    float *dat;
    int i;
    float datamin, datamax, datamean;

    /*
     * generate random data
     */

    dat = (float *)malloc(nx * sizeof(float));
    srand(0);
    for (i=0;i<nx;i++) {
        dat[i] = 2*((float)rand()/RAND_MAX)-1.;
    }

    /*
     * find min/mean/max
     */ 

    datamin = 1e+19;
    datamax =-1e+19;
    datamean = 0;

    
    for (i=0;i<nx;i++) {
        if (dat[i] < datamin) datamin=dat[i];
        if (dat[i] > datamax) datamax=dat[i];
        datamean += dat[i];
    }
    datamean /= nx;
    free(dat);

    printf("Min/mean/max = %f,%f,%f\n", datamin,datamean,datamax);

    return 0;
}
